soh/src/overlays/actors/ovl_En_Hy/z_en_hy.c

```diff
@@ -438,7 +438,7 @@ u16 func_80A6F810(PlayState* play, Actor* thisx) {
 
                 if (followingDog != 0) {
                     this->unk_215 = false;
-                    return (followingDog == 1) ? 0x709F : 0x709E;
+                    return ((followingDog == 1) || (CVarGetInteger("gAllDogsRichard", 0) && followingDog != 0)) ? 0x709F : 0x709E;
```

Comment:
Isn't ` && followingDog != 0` unnecessary

Reply:
I think `following dog == 1` is Richard specifically. The `followingDog != 0` is modifying the cvar, thus looking for the cvar AND *any* dog following.

Reply:
Oh, I see it now.

Reply:
Sure is. It was what I used to test is that was the right code and it didn't occur to me that just the cvar being set accomplished the same thing. Fixed

Reply:
It doesn't get here if followingdog is 0 in the first place, and the cvar is supposed to already bypass what dog it is, so why would there be another check with the cvar enabled?

Reply:
That's why I said "I see it now"

Reply:
I was typing while you posted the comment, so I only saw it afterwards :P

Reply:
Thanks for pointing that out @aMannus